server/src/main/java/org/candlepin/dto/api/v1/EnvironmentDTO.java

```diff
@@ -0,0 +1,408 @@
+     *
+     * @return a reference to this DTO object.
+     */
+    public EnvironmentDTO setEnvironmentContent(Collection<EnvironmentContentDTO> environmentContent) {
+        if (environmentContent != null) {
+            if (this.environmentContent == null) {
+                this.environmentContent = new HashMap<String, EnvironmentContentDTO>();
+            }
+            else {
+                this.environmentContent.clear();
+            }
+
+            for (EnvironmentContentDTO dto : environmentContent) {
+                if (dto == null || dto.getContent() == null ||
+                    StringUtils.isEmpty(dto.getContent().getId())) {
+                    throw new IllegalArgumentException("environment content is null or incomplete");
+                }
+
+                this.environmentContent.put(dto.getContent().getId(), dto);
+            }
+        }
+        else {
+            this.environmentContent = null;
+        }
+        return this;
+    }
+
+    /**
+     * Adds the given content to this environment DTO. If a matching content has already been added to
+     * this environment, it will be overwritten by the specified content.
+     *
+     * @param dto
+     *  The environment content DTO to add to this environment
+     *
+     * @throws IllegalArgumentException
+     *  if content is null or incomplete
+     *
+     * @return
+     *  true if adding the content resulted in a change to this product; false otherwise
+     */
+    public boolean addEnvironmentContent(EnvironmentContentDTO dto) {
+        if (dto == null) {
+            throw new IllegalArgumentException("dto is null");
+        }
+
+        if (dto.getContent() == null || dto.getContent().getId() == null) {
+            throw new IllegalArgumentException("dto references incomplete content");
+        }
+
+        boolean changed = false;
+        boolean matched = false;
+        String contentId = dto.getContent().getId();
+
+        if (this.environmentContent == null) {
+            this.environmentContent = new HashMap<String, EnvironmentContentDTO>();
+            changed = true;
+        }
+        else {
+            EnvironmentContentDTO existing = this.environmentContent.get(dto.getContent().getId());
+            changed = !dto.equals(existing);
+        }
+
+        if (changed) {
+            this.environmentContent.put(dto.getContent().getId(), dto);
+        }
+
+        return changed;
+    }
+
+    /**
+     * Adds the given content to this DTO. If a matching content has already been added to
+     * this environment, it will be overwritten by the specified content.
+     *
+     * @param dto
+     *  The product content DTO to add to this product
+     *
+     * @throws IllegalArgumentException
+     *  if content is null
+     *
+     * @return
+     *  true if adding the content resulted in a change to this product; false otherwise
+     */
+    public boolean addContent(ContentDTO dto, boolean enabled) {
```

Comment:
This needs to accept a null enabled value, as this field is used as an environment override only if it's not null.